packages/extension-ui/src/util/getFaviconUrl.ts

```diff
@@ -0,0 +1,14 @@
+// Copyright 2019-2023 @polkadot/extension-ui authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+export function getFaviconUrl(url: string): string {
+  try {
+    const domain = new URL(url).hostname;
+
+    return `https://www.google.com/s2/favicons?domain=${domain}`;
```

Comment:
To be honest I would prefer JS solution, getting origin from domain but use JS code as if google at some point removes the endopint, or modifies how the query is created the whole wallet will need to be updated just because of that